gems/aws-sdk-core/spec/spec_helper.rb

```diff
@@ -12,14 +12,33 @@
 require_relative 'support/credentials_provider'
 require_relative 'support/refreshing_credentials_provider'
 
+RSpec.configure do |config|
+  config.before(:each) do
+    # Clear the current ENV to avoid loading credentials.
+    # This was previously mocked with stub_const but was provided a hash.
+    ENV.clear
+
+    # disable loading credentials from shared file
+    allow(Dir).to receive(:home).and_raise(ArgumentError)
+
+    # disable instance profile credentials
+    token_path = '/latest/api/token'
+    path = '/latest/meta-data/iam/security-credentials/'
+    stub_request(:get, "http://169.254.169.254#{path}").to_raise(SocketError)
+    stub_request(:put, "http://169.254.169.254#{token_path}").to_raise(SocketError)
```

Comment:
Is this necessary for every test?

Reply:
This is what we do in V3 to prevent tests from trying to load credentials

Reply:
Maybe when we do stub responses, we can prevent credential chain from going too?

Reply:
We do.  

The issue is in clients that are created in specs without stub_responses.  This happens for example when testing credential providers that create an STS client or some such.  The client that ends up created may not be created using stub_responses (because the code in the class shouldn't have that...).  

Having this code helps catch test failures that happen in CI, but not locally (where you probably do have some source of credentials defined)